File: mobile/src/main/java/com/rndapp/roostremote/models/BoolOption.java
package com.rndapp.roostremote.models;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by ell on 5/16/15.
 */
public class BoolOption extends Option{
    protected String name;
    protected boolean value;

    public BoolOption(String name, boolean value) {
        this.name = name;
        this.value = value;
    }

    @Override
    public JSONObject getJsonObject(String key) throws JSONException {
        JSONObject jsonObject = new JSONObject();
        jsonObject.put(key, value);
        return jsonObject;
    }

    public String getName() {
        return name;
    }

    public boolean getValue(){
        return value;
    }
}

File: mobile/src/main/java/com/rndapp/roostremote/models/BoolOption.kt
package com.rndapp.roostremote.models

import android.util.Log

import org.json.JSONException
import org.json.JSONObject

/**
 * Created by ell on 5/16/15.
 */
class BoolOption(name: String, value: Boolean) : Option(name, value) {

    @Throws(JSONException::class)
    override fun getJsonObject(key: String): JSONObject {
        val jsonObject = JSONObject()
        jsonObject.put(key, value as Boolean)
        Log.d("bool opt", jsonObject.toString())
        return jsonObject
    }
}
